ovn_k8s/modes/overlay.py

```diff
@@ -436,3 +480,380 @@ def sync_services(self, services):
+
+        Args:
+            namespace(:obj:`str`): The policy's namespace
+            policy_data(dict): Policy info as returned by the kubernetes
+                API server, contains both the spec and status sections
+
+        """
+        for rule in policy_data['spec']['ingress']:
+            address_set = self._build_rule_address_set_name(
+                policy_data['metadata']['name'],
+                namespace,
+                rule)
+            # First verify that the address set does not already exist
+            if not ovn_nbctl("--data=bare", "--no-heading",
+                             "--columns=_uuid", "find",
+                             "address_set", "name=%s" % address_set):
+                ovn_nbctl('create', 'address_set', 'name=%s' % address_set)
+
+    def destroy_address_set(self, namespace, policy_data):
+        for rule in policy_data['spec']['ingress']:
+            address_set = self._build_rule_address_set_name(
+                policy_data['metadata']['name'],
+                namespace,
+                rule)
+            if ovn_nbctl("--data=bare", "--no-heading",
+                         "--columns=_uuid", "find",
+                         "address_set", "name=%s" % address_set):
+                ovn_nbctl('destroy', 'address_set', address_set)
+
+    def add_to_address_set(self, pod_ip, namespace, policy_data):
+        for rule in policy_data['spec']['ingress']:
+            rule_address_set = self._build_rule_address_set_name(
+                policy_data['metadata']['name'],
+                namespace,
+                rule)
+            ovn_nbctl('--if-exists', 'add', 'address_set',
+                      rule_address_set, 'addresses', pod_ip)
+
+    def remove_from_address_set(self, pod_ip, namespace, policy_data):
+        for rule in policy_data['spec']['ingress']:
+            rule_address_set = self._build_rule_address_set_name(
+                policy_data['metadata']['name'],
+                namespace,
+                rule)
+            ovn_nbctl('--if-exists', 'remove', 'address_set',
+                      rule_address_set, 'addresses', pod_ip)
+
+    def build_pseudo_acls(self, policy_data):
+        # Build pseudo ACL list for policy
+        policy_id = policy_data['metadata']['uid']
+        policy_pseudo_acls = []
+        for rule in policy_data['spec']['ingress']:
+            ports_data = rule.get('ports', [])
+            protocol_ports = []
+            for item in ports_data:
+                protocol_ports.append((item['protocol'], item.get('port')))
+            from_data = rule.get('from')
+            if not from_data:
+                src_pod_ips = '*'
+            else:
+                rule_address_set = self._build_rule_address_set_name(
+                    policy_data['metadata']['name'],
+                    policy_data['metadata']['namespace'],
+                    rule)
+                src_pod_ips = 'address_set(%s)' % rule_address_set
```

Comment:
I don't think this will work. According to https://github.com/openvswitch/ovs/commit/ea382567e816ac3bdb6365fdec90438156cf51d1 this line should be written down as:
`src_pod_ips = '$%s' % rule_address_set`